ambry-store/src/test/java/com.github.ambry.store/StoreDescriptorTest.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.store;
+
+import java.io.DataOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.UUID;
+import junit.framework.Assert;
+import org.junit.Test;
+
+
+/**
+ * Tests {@link StoreDescriptor}
+ */
+public class StoreDescriptorTest {
+
+  /**
+   * Tests {@link StoreDescriptor} for unit tests for instantiation and converting bytes into StoreDescriptor
+   * @throws IOException
+   */
+  @Test
+  public void testStoreDescriptor() throws IOException {
+    String tempDirPath = "/tmp/";
+    File storeDescriptorFile = new File(tempDirPath + "/" + StoreDescriptor.STORE_DESCRIPTOR_FILENAME);
+    storeDescriptorFile.delete();
+
+    StoreDescriptor storeDescriptor = new StoreDescriptor(tempDirPath);
+    // store descriptor file should have been created.
+    StoreDescriptor newStoreDescriptor = new StoreDescriptor(tempDirPath);
+    Assert.assertEquals("IncarnationId mismatch ", storeDescriptor.getIncarnationId(),
+        newStoreDescriptor.getIncarnationId());
+
+    // Create StoreDescriptor file with new incarnationId
+    UUID incarnationIdUUID = UUID.randomUUID();
+    int size = StoreDescriptor.VERSION_SIZE +
+        StoreDescriptor.INCARNATION_ID_LENGTH_SIZE + incarnationIdUUID.toString().getBytes().length;
+    byte[] toBytes = new byte[size];
+    ByteBuffer byteBuffer = ByteBuffer.wrap(toBytes);
+    byteBuffer.putShort(StoreDescriptor.VERSION_0);
+    byteBuffer.putInt(incarnationIdUUID.toString().getBytes().length);
+    byteBuffer.put(incarnationIdUUID.toString().getBytes());
+    byteBuffer.flip();
+
+    storeDescriptorFile = new File(tempDirPath + "/" + StoreDescriptor.STORE_DESCRIPTOR_FILENAME);
+    storeDescriptorFile.createNewFile();
+    DataOutputStream dataOutputStream = new DataOutputStream(new FileOutputStream(storeDescriptorFile));
+    dataOutputStream.write(toBytes);
+    dataOutputStream.close();
+
+    storeDescriptor = new StoreDescriptor(tempDirPath);
+    Assert.assertEquals("IncarnationId mismatch ", incarnationIdUUID, storeDescriptor.getIncarnationId());
```

Comment:
in the previous lines, we have tested only the case where incarnationId was generated by the StoreDescriptor itself. Here, want to check if deserialization of the file works as expected.